File: poker-webapp/src/main/webapp/controls/Card.js
sap.ui.define([ "sap/ui/core/Control" ], function(Control) {
	"use strict";

	var toUnicode = function(suit, kind) {

		// See here for unicodes
		// https://en.wikipedia.org/wiki/Playing_cards_in_Unicode
		// insert in your view:
		// <poker:Card class="sapUiSmallMarginBeginEnd" suit="spades" kind="10"/>

		var unicode = "&#x1F0"
		if (suit === "spades") {
			unicode = unicode + "A";
		} else if (suit === "hearts") {
			unicode = unicode + "B";
		} else if (suit === "diamonds") {
			unicode = unicode + "C";
		} else if (suit === "clubs") {
			unicode = unicode + "D";
		}
		if (kind >= 2 && kind <= 9) {
			unicode = unicode + kind;
		} else {
			if (kind === "10") {
				unicode = unicode + "A";
			} else if (kind === "jack") {
				unicode = unicode + "B";
			} else if (kind === "queen") {
				unicode = unicode + "D";
			} else if (kind === "king") {
				unicode = unicode + "E";
			} else if (kind === "ace") {
				unicode = unicode + "1";
			}
		}

		if (suit === "default" || kind === "default") {
			unicode = unicode + "F1";
		}

		return unicode;

	};

	var getFontColor = function(suit) {
		if (suit === "hearts" || suit === "diamonds") {
			return "red";
		}
		return "black";
	}

	return Control.extend("poker.controls.Card", {
		metadata : {
			properties : {
				suit : {
					type : "string",
					defaultValue : "default"
				},
				kind : {
					type : "string",
					defaultValue : "default"
				}
			},
			aggregations : {

			},
			events : {

			}
		},
		init : function() {

		},
		renderer : function(oRM, oControl) {
			oRM.write("<div");
			oRM.writeControlData(oControl);
			oRM.writeClasses();
			if (getFontColor(oControl.getSuit()) === "red") {
				oRM.write("><div style=\"font-size:200px;color:red;\">");
			} else {
				oRM.write("><div style=\"font-size:200px;color:black;\">");
			}
			oRM.write(toUnicode(oControl.getSuit(), oControl.getKind()));
			oRM.write("</div></div></br></br>");
		}
	});
});

File: poker-webapp/src/main/java/com/sap/ase/poker/gameguide/models/Card.java
package com.sap.ase.poker.gameguide.models;

public class Card implements Comparable<Card> {

	public enum Suits {
		DIAMONDS, HEARTS, SPADES, CLUBS;
	}

	public enum Kinds {
		TWO(1, "2"), THREE(2, "3"), FOUR(3, "4"), FIFE(4, "5"), SIX(5, "6"), SEVEN(6, "7"), EIGHT(7, "8"), NINE(8,
				"9"), TEN(9, "10"), JACK(10, "Jack"), QUEEN(11, "Queen"), KING(12, "King"), ACE(13, "Ace");

		public int rank;
		public String value;

		private Kinds(int rank, String value) {
			this.rank = rank;
			this.value = value;
		}
	}

	private Card.Kinds kind;
	private Card.Suits suit;

	public Card() {
		super();
	}

	public Card(Kinds kind, Suits suit) {
		super();
		this.kind = kind;
		this.suit = suit;
	}

	public Card.Kinds getKind() {
		return kind;
	}

	public Card.Suits getSuit() {
		return suit;
	}

	public void setKind(Card.Kinds kind) {
		this.kind = kind;
	}

	public void setSuit(Card.Suits suit) {
		this.suit = suit;
	}

	@Override
	public int compareTo(Card c) {
		return c.getKind().rank - this.kind.rank;
	}
}
